server/segment_rpc.go

```diff
@@ -549,3 +555,11 @@ func genPayment(sess *BroadcastSession, numTickets int) (string, error) {
 
 	return base64.StdEncoding.EncodeToString(data), nil
 }
+
+func validatePrice(sess *BroadcastSession) bool {
+	oPrice := big.NewRat(sess.OrchestratorInfo.PriceInfo.GetPricePerUnit(), sess.OrchestratorInfo.PriceInfo.GetPixelsPerUnit())
```

Comment:
I think we should make sure we don't trigger a division by zero error here - `sess.OrchestratorInfo.PriceInfo` could be anything here since we do not apply any checks on a new `OrchestratorInfo` message when calling `updateOrchestratorInfo()` in `SubmitSegment()`

Reply:
While that is true we only call `updateOrchestratorInfo` when `TranscodeResult.OrchestratorInfo` is not `nil`. 

Since `PriceInfo` is always set on the Orchestrator (it can never be nil) there should always be a  `PriceInfo` field included in `TranscodeResult.OrchestratorInfo`. If that's not the case something went horribly wrong in creating the HTTP response.

We could add a guard clause here but if the assumption above doesn't hold up it means code elsewhere isn't working as intended. So I felt like a panic there in such a case would be "OK" as it is unexpected behaviour. 

Wdyt?

Reply:
for completeness since I tend to forget about offchain mode: 

`PriceInfo` can be nil on the orchestrator if it's in offchain mode, but then we will never be calling `genPayment` anyway.

Reply:
> Since PriceInfo is always set on the Orchestrator (it can never be nil) there should always be a PriceInfo field included in TranscodeResult.OrchestratorInfo. If that's not the case something went horribly wrong in creating the HTTP response.

While we know what the default reference O implementation will be, we shouldn't assume that all Os will be using the same implementation. B shouldn't crash just because an O it is working with is using a faulty implementation (purposeful or not).

Reply:
Fair argument !

Reply:
Fixed in https://github.com/livepeer/go-livepeer/pull/992/commits/d828e873a90affa1a5b8f4544df6b4f0103d68e7

Tests added in https://github.com/livepeer/go-livepeer/pull/992/commits/f7cc31f5c8f88b98fe62df5ad649f5be3a37ed98